nav2_controller/src/nav2_controller.cpp

```diff
@@ -305,22 +306,70 @@ void ControllerServer::setPlannerPath(const nav_msgs::msg::Path & path)
     end_pose.pose.position.x, end_pose.pose.position.y);
 }
 
+double getDistance(
+  const geometry_msgs::msg::PoseStamped & pose_a,
+  const geometry_msgs::msg::PoseStamped & pose_b)
+{
+  double x_diff = fabs(pose_a.pose.position.x - pose_b.pose.position.x);
+  double y_diff = fabs(pose_a.pose.position.y - pose_b.pose.position.y);
+
+  return std::hypot(x_diff, y_diff);
+}
+
+geometry_msgs::msg::TwistStamped getTwistToPose(
```

Comment:
Why is this here?

Reply:
All of these changes in the controller server - I'm not sure they belong here. Can you explain this?